DMX.Core.Api/Controllers/LabsController.cs

```diff
@@ -85,5 +86,47 @@ public async ValueTask<ActionResult<List<Lab>>> GetAllLabsAsync()
                 return InternalServerError(labOrchestrationServiceException);
             }
         }
+
+        [HttpDelete("{labId}")]
+#if RELEASE
+        [RequiredScope(RequiredScopesConfigurationKey = "AzureAd:Scopes:DeleteLab")] 
+#endif
+        public async ValueTask<ActionResult<Lab>> DeleteLabByIdAsync(Guid labId)
+        {
+            try
+            {
+                Lab deletedLab =
+                    await this.labOrchestrationService.RemoveLabByIdAsync(labId);
+
+                return Ok(deletedLab);
+            }
+            catch (LabValidationException labValidationException)
+                when (labValidationException.InnerException is NotFoundLabException)

```

Comment:
@hassanhabib Taarafo.Core uses "NotFoundPostException" when the guid cannot be matched against a database item, however the LabService uses "NullLabException".

I'm partial to using "NotFoundLabException" since I think it better describes the circumstance. Thoughts?

Reply:
I agree. 100%. great catch @mimendel